File: DigiMox/app/src/main/java/com/digimox/adapters/DMCategorySubAdapter.java
package com.digimox.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.digimox.R;
import com.digimox.fragments.DMMainCategorySubFragment;
import com.digimox.models.response.DMMainCategory;
import com.digimox.utils.DMUtils;

import java.util.ArrayList;

/**
 * Created by Deepesh on 23-Aug-15.
 */
public class DMCategorySubAdapter extends RecyclerView.Adapter<DMCategorySubAdapter.ViewHolder> {
    private final TypedValue mTypedValue = new TypedValue();
    private Context context;
    private OnItemClickListener mItemClickListener;
    private DMMainCategorySubFragment dmMainCategorySubFragment;
    private ArrayList<DMMainCategory> dmMainCategories;

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        private ImageView categoryImage;
        private TextView categoryName;
        private ProgressBar mainCategoryProgress;
        private View mView;

        public ViewHolder(View view) {
            super(view);
            this.mView = view;
            view.setOnClickListener(this);
            categoryImage = (ImageView) view.findViewById(R.id.categoryImage);
            categoryName = (TextView) view.findViewById(R.id.category_name);
            mainCategoryProgress = (ProgressBar) view.findViewById(R.id.progress_main_category);

        }

        @Override
        public void onClick(View view) {
            if (mItemClickListener != null) {
                mItemClickListener.onItemClick(view, getPosition());
            }
        }
    }


    public interface OnItemClickListener {
        public void onItemClick(View view, int position);
    }

    public void setOnItemClickListener(final OnItemClickListener mItemClickListener) {
        this.mItemClickListener = mItemClickListener;
    }

    public DMCategorySubAdapter(Context context, DMMainCategorySubFragment dmMainCategorySubFragment, ArrayList<DMMainCategory> dmMainCategories) {
        context.getTheme().resolveAttribute(R.attr.selectableItemBackground, mTypedValue, true);
        this.context = context;
        this.dmMainCategorySubFragment = dmMainCategorySubFragment;
        this.dmMainCategories = dmMainCategories;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater mInflater = LayoutInflater.from(parent.getContext());
        ViewGroup defaultGroup = (ViewGroup) mInflater.inflate(R.layout.category_list_item, parent, false);
        return new ViewHolder(defaultGroup);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, final int position) {
        final DMMainCategory dmMainCategory = dmMainCategories.get(position);
        if (!TextUtils.isEmpty(dmMainCategory.getGroupImage())) {
            DMUtils.setImageUrlToView(context, holder.categoryImage, dmMainCategory.getGroupImage(), holder.mainCategoryProgress);
        }
        holder.categoryName.setText(dmMainCategory.getGroupName());
        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                goToSubCategory(dmMainCategory, position);
            }
        });

    }

    private void goToSubCategory(DMMainCategory dmMainCategory, int position) {
        dmMainCategorySubFragment.clickMainCategory(dmMainCategories, dmMainCategory, position);
    }

    @Override
    public int getItemCount() {
        return dmMainCategories.size();
    }

}

File: DigiMox/app/src/main/java/com/digimox/adapters/DMCategorySubAdapter.kt
package com.digimox.adapters

import android.content.Context
import android.text.TextUtils
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.digimox.R
import com.digimox.fragments.DMMainCategorySubFragment
import com.digimox.models.response.DMMainCategory
import com.digimox.utils.DMUtils
import java.util.*

/**
 * Created by Deepesh on 23-Aug-15.
 */
class DMCategorySubAdapter(private val context: Context, private val dmMainCategorySubFragment: DMMainCategorySubFragment, private val dmMainCategories: ArrayList<DMMainCategory>) : RecyclerView.Adapter<DMCategorySubAdapter.ViewHolder>() {
    private val mTypedValue = TypedValue()
    private var mItemClickListener: OnItemClickListener? = null

    inner class ViewHolder(val mView: View) : RecyclerView.ViewHolder(mView), View.OnClickListener {
        val categoryImage: ImageView
        val categoryName: TextView
        val mainCategoryProgress: ProgressBar

        init {
            mView.setOnClickListener(this)
            categoryImage = mView.findViewById<View>(R.id.categoryImage) as ImageView
            categoryName = mView.findViewById<View>(R.id.category_name) as TextView
            mainCategoryProgress = mView.findViewById<View>(R.id.progress_main_category) as ProgressBar

        }

        override fun onClick(view: View) {
            if (mItemClickListener != null) {
                mItemClickListener!!.onItemClick(view, position)
            }
        }
    }


    interface OnItemClickListener {
        fun onItemClick(view: View, position: Int)
    }

    fun setOnItemClickListener(mItemClickListener: OnItemClickListener) {
        this.mItemClickListener = mItemClickListener
    }

    init {
        context.theme.resolveAttribute(R.attr.selectableItemBackground, mTypedValue, true)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val mInflater = LayoutInflater.from(parent.context)
        val defaultGroup = mInflater.inflate(R.layout.category_list_item, parent, false) as ViewGroup
        return ViewHolder(defaultGroup)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val dmMainCategory = dmMainCategories[position]
        if (!TextUtils.isEmpty(dmMainCategory.groupImage)) {
            DMUtils.setImageUrlToView(context, holder.categoryImage, dmMainCategory.groupImage, holder.mainCategoryProgress)
        }
        holder.categoryName.text = dmMainCategory.groupName
        holder.mView.setOnClickListener { goToSubCategory(dmMainCategory, position) }

    }

    private fun goToSubCategory(dmMainCategory: DMMainCategory, position: Int) {
        dmMainCategorySubFragment.clickMainCategory(dmMainCategories, dmMainCategory, position)
    }

    override fun getItemCount(): Int {
        return dmMainCategories.size
    }

}

